### PR TITLE
#484: Keep tempered algorithm in sync with implementation in vt

### LINKED ISSUE #484: Keep tempered algorithm in sync with implementation in vt
This is because [vt#2201](https://github.com/DARMA-tasking/vt/issues/2201) will result in a version of the tempered algorithm (and transfer strategies) not necessarily synchronized -- or with some improvements.

### PR DESCRIPTION
**Pull request by ppebay**
resolves #484

### CODE REVIEW

**Comment by cwschilly** [comment]
@ppebay This PR now passes all tests

File: src/lbaf/IO/lbsConfigurationValidator.py [hunk 1]
```diff
@@ -159,7 +159,11 @@ def __init__(self, config_to_validate: dict, logger: Logger):
                          Optional("cluster_swap_rtol"): And(
                             float,
                             lambda x: x > 0.0,
-                            error="Should be of type 'float' and magnitude > 0.0"),
+                            error="Should be of type 'float' and > 0.0"),
+                         Optional("max_subclusters"): And(
+                            int,
+                            lambda x: x > 0.0,
+                            error="Should be of type 'int' and > 0"),
```
Comment (cwschilly) [inline src/lbaf/IO/lbsConfigurationValidator.py:166]: Should we add a unit test for this new parameter?

File: src/lbaf/Execution/lbsTransferStrategyBase.py [hunk 2]
```diff
@@ -65,8 +65,8 @@ def _get_ranks_to_traverse(self, ranks: list, known_peers: list):
         # Iterate over all provided ranks
         for r_src in ranks:
             # Ignore ranks without migratable objects
-            if not r_src.get_migratable_objects():
-                continue
+            #if not r_src.get_migratable_objects():
+            #    continue
```
Comment (cwschilly) [inline src/lbaf/Execution/lbsTransferStrategyBase.py:69]: Can these lines be deleted?

**Review by cwschilly — COMMENTED** [review]
I can write a unit test for the `max_clusters` parameter if it would be helpful--just let me know

File: src/lbaf/IO/lbsConfigurationValidator.py (hunk 1 above)
Comment (ppebay) [reply src/lbaf/IO/lbsConfigurationValidator.py:166]: Yes please, that would be good

File: src/lbaf/Execution/lbsTransferStrategyBase.py (hunk 2 above)
Comment (ppebay) [reply src/lbaf/Execution/lbsTransferStrategyBase.py:69]: Oh no -- please remove the comment `#`s
This is also a leftover debugging hack. But indeed if no objects are migratable the rank should be skipped

File: src/lbaf/IO/lbsConfigurationValidator.py (hunk 1 above)
Comment (cwschilly) [reply src/lbaf/IO/lbsConfigurationValidator.py:166]: Ok, I added the tests

**Review by cwschilly — APPROVED** [review]
LGTM

**Comment by ppebay** [comment]
@cwschilly  When undefined, the new parameter is consider to be infinite (before that addition, there was a hard-coded value of 10 in the code). When the value is not provided, it's going to be considered infinite -- which in turn might cause the code to go into a computationally intractable cluster swap exploration. 

As a result other CI tests might have come unfeasible unless the parameter is explicitly added to their configuration file.

**Review by cwschilly — APPROVED** [review]
The CI is hanging occasionally, but I have not been able to find anything in this PR that causes it. Any hung jobs will pass on a rerun.

**Pull request merged**